python/sglang/srt/server.py

```diff
@@ -450,13 +450,15 @@ def launch_engine(
         for tp_rank in tp_rank_range:
             reader, writer = mp.Pipe(duplex=False)
             gpu_id = server_args.base_gpu_id + tp_rank % tp_size_per_node
+            os.environ["CUDA_VISIBLE_DEVICES"] = str(gpu_id)
```

Comment:
Does this work for AMD? We might need to find the correct env vars for AMD as well.
cc @HaiShaw

Reply:
Perhaps not; the tests pass for me locally but don't seem to pass in CI.

Reply:
According to these docs, this should work: https://rocm.docs.amd.com/en/latest/conceptual/gpu-isolation.html#cuda-visible-devices